src/js/rtc_session.js

```diff
@@ -880,7 +880,9 @@ export default class RtcSession {
         self._pc.onicecandidate = hitch(self, self._onIceCandidate);
         self._pc.oniceconnectionstatechange = hitch(self, self._onIceStateChange);
 
-        self._isLegacyStatsReportSupported();
+        self._isLegacyStatsReportSupported(function(result) {
+            self._legacyStatsReportSupport = result;
```

Comment:
Theoretically this creates a race condition. One way to resolve this is to make _isLegacyStatsReportSupported returning a promise of boolean, here in connect() you can chain the latter process to avoid racing.
```
isLegacyStatsReportSupported().then(result => {
  self._legacyStatsReportSupport = result;
  self.transit(new GrabLocalMediaState(self));
});
```
Also consider moving _isLegacyStatsReportSupported to some other place (utils)
(practically it probably won't happen)

Reply:
Removed race condition and moved isLegacyStatsReportSupported to utils.